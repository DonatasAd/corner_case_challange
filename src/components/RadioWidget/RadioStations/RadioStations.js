import React, { useEffect } from 'react';
import styles from './RadioStations.module.css';
import RadioStation from './RadioStation/RadioStation';
import { connect } from 'react-redux';
import * as radioStationsActions from '../../../store/actions/radioStations';
import PropTypes from 'prop-types';

export function RadioStations({ fetchRadiosStations, radioStations }) {
  useEffect(() => {
    fetchRadiosStations();
  }, [fetchRadiosStations]);
  return (
    <div className={styles.Root}>
      <ul className={styles.List}>
        {radioStations && radioStations.map((el, index) => <RadioStation radio={el} key={el.frequency} appendLine={index !== radioStations.length - 1} />)}
      </ul>
    </div>
  );
}

RadioStations.propTypes = {
  radioStations: PropTypes.array,
  fetchRadiosStations: PropTypes.func,
};

const mapDispatchToProps = {
  fetchRadiosStations: radioStationsActions.fetchRadiosStations,
};

const mapStateToProps = (state) => ({
  radioStations: state.radioStation.radioStations,
});

export default connect(mapStateToProps, mapDispatchToProps)(RadioStations);
