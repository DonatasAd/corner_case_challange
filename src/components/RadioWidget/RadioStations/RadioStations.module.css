.Root {
  flex: 1 0 20%;
  overflow-y: scroll;
  -ms-overflow-style: none; /* IE and Edge */
  scrollbar-width: none; /* Firefox */
}

.Root::-webkit-scrollbar {
  display: none;
}

.List {
  padding: 0rem;
  margin: 1rem;
}
