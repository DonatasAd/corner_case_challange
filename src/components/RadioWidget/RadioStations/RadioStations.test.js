import React from 'react';
import { shallow } from 'enzyme';
import { RadioStations } from './RadioStations';

describe('RadioStations', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<RadioStations />);
    expect(component).toMatchSnapshot();
  });
  it('should render two <li>', () => {
    let component = shallow(<RadioStations />);
    component.setProps({
      radioStations: [
        { name: 'test2', frequency: 1 },
        { name: 'test2', frequency: 2 },
      ],
    });
    expect(component.find('ul').children()).toHaveLength(2);
  });
});
