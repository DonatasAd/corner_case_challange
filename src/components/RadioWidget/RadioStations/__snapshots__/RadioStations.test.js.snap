// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioStations should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
